### PR TITLE
NETOBSERV-1610: document cardinality warning per field

### PR DESCRIPTION
**Pull request by jotak**
## Description

- Add `cardinalityWarn` property to frontend fields config
- Add test checking all fields are correctly flagged
- Add column to generated doc of flows format

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [x] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Comment by jotak** [comment]
@OlivierCazade I guess this could also be leveraged in https://issues.redhat.com/browse/NETOBSERV-1447 ?
Does my list of high cardinality fields matches what you had?

@skrthomas : in https://github.com/openshift/openshift-docs/pull/74412 : I guess in "Configuring custom metrics" we should explain to refer to this fields list to check what is fine to use (mentioned as "fine" in fields list), or to use with care (mentioned as "careful"), or that cannot be used (mentioned as "avoid")

File: docs/flows-format.adoc [hunk 1]
```diff
@@ -9,105 +9,132 @@ The "Filter ID" column shows which related name to use when defining Quick Filte
 
 The "Loki label" column is useful when querying Loki directly: label fields need to be selected using link:https://grafana.com/docs/loki/latest/logql/log_queries/#log-stream-selector[stream selectors].
 
+The "Cardinality" column gives information about the implied metric cardinality if this field was to be used as a Prometheus label with the `FlowMetrics` API. Refer to the `FlowMetrics` documentation for more information on using this API.
```
Comment (jotak) [inline docs/flows-format.adoc:12]: cc @skrthomas for reviewing this part

**Review by jpinsonneau — APPROVED** [review]
Looks good, thanks @jotak

File: docs/flows-format.adoc (hunk 1 above)
Comment (skrthomas) [inline docs/flows-format.adoc:12]:
```suggestion
The "Cardinality" column gives information about the implied metric cardinality if this field is used as a Prometheus label with the `FlowMetrics` API. Refer to the `FlowMetrics` documentation for more information on using this API.
```

**Review by memodi — COMMENTED** [review]
@OlivierCazade where should these warnings appear?

**Pull request merged**